packages/template-retail-react-app/app/pages/checkout/index.test.js

```diff
@@ -511,63 +359,190 @@ test.skip('Can proceed through checkout as registered customer', async () => {
     // Place the order
     user.click(placeOrderBtn)
 
+    // Should now be on our mocked confirmation route/page
+    expect(await screen.findByText(/success/i)).toBeInTheDocument()
+})
+
+test('Can edit address during checkout as a registered customer', async () => {
+    await logInDuringCheckout()
+
+    const firstAddress = screen.getByTestId('sf-checkout-shipping-address-0')
+    user.click(within(firstAddress).getByText(/edit/i))
+
+    // Wait for the edit address form to render
+    await waitFor(() =>
+        expect(screen.getByTestId('sf-shipping-address-edit-form')).not.toBeEmptyDOMElement()
+    )
+
+    expect(screen.getByLabelText(/first name/i)).toBeInTheDocument()
+
+    // Edit and save the address
+    user.clear(screen.getByLabelText('Address'))
+    user.type(screen.getByLabelText('Address'), '369 Main Street')
+    user.click(screen.getByText(/save & continue to shipping method/i))
+
+    // Wait for next step to render
     await waitFor(() => {
-        expect(window.location.pathname).toEqual('/uk/en-GB/checkout/confirmation')
+        expect(screen.getByTestId('sf-toggle-card-step-2-content')).not.toBeEmptyDOMElement()
     })
+
+    expect(screen.getByText('369 Main Street')).toBeInTheDocument()
 })
 
-// @TODO: fix in hook integration checkout address ticket
-test.skip('Can edit address during checkout as a registered customer', async () => {
-    jest.setTimeout(30000)
+test('Can add address during checkout as a registered customer', async () => {
+    await logInDuringCheckout()
+
+    global.server.use(
+        rest.post('*/customers/:customerId/addresses', (req, res, ctx) => {
+            return res(ctx.delay(0), ctx.status(200), ctx.json(req.body))
+        })
+    )
+
+    // Add address
+    user.click(screen.getByText(/add new address/i))
+
+    const firstName = await screen.findByLabelText(/first name/i)
+    user.type(firstName, 'Test2')
+    user.type(screen.getByLabelText(/last name/i), 'McTester')
+    user.type(screen.getByLabelText(/phone/i), '7275551234')
```

Comment:
This block of code was calling `byRole` queries a lot. I replaced them with `byLabelText` to speed them up, as we've learned recently.